internal/stats/metrics.go

```diff
@@ -15,6 +15,12 @@ type BaseMetric struct {
 	Attributes map[string]any `json:"attributes,omitempty"`
 }
 
+type CourseAssignmentEmailMetric struct {
+	CourseID     string `json:"course,omitempty"`
+	AssignmentID string `json:"assignment,omitempty"`
+	UserEmail    string `json:"user,omitempty"`
+}
```

Comment:
Abstracting this out seemed to be a pretty big pain to several places that created stats (or queries).

Of course, abstraction is usually a good thing, because it makes our future lives easier.
But here, I'm not sure what we gained from it.
It was embedded into a few structs, but then never pulled out of the struct for commons use, e.g.,
```
someFunc(myCourseMetric.CourseAssignmentEmailMetric)
someFunc(myAPIRequestMetric.CourseAssignmentEmailMetric)
```

I would think about it and see if you think we should keep the abstraction or just put the fields in directly.